docs/DeveloperGuide.md

```diff
@@ -604,42 +604,230 @@ testers are expected to do more *exploratory* testing.
+      Expected: No person is added. Error details shown in the status message.
+   
+   3. Other incorrect add commands to try: `add`, `add ...` (where `...` is any pair of prefix and value)<br>
       Expected: Similar to previous.
 
-1. _{ more test cases …​ }_
+### Editing a contact
+1. Editing a contact field that is shown in the list
+   1. Test case: `edit 1 name/John Tan`
+      Expected: Contact shown at index 1 has name edited to `John Tan`. Details shown in status message.
+   
+   2. Test case: `edit 0 name/John Tan`
+      Expected: No contact is edited. Details of error shown in status message.
+
+   3. Other incorrect edit commands to try: `edit -1 name/John Tan`, `edit x name/John Tan` (where `x` is a number that is not in the shown list)
+      Expected: Similar to previous
+
+### Adding grade to a person
+1. Add grades to a person
+   1. Prerequisites: 
+      1. `assignment.json` in `data` with the following content:
+        ```json
+        {
+          "assignments" : [
+            {
+              "name": "Ex01",
+              "maxScore": 5
+            },{
+              "name": "Ex02",
+              "maxScore": 5
+            },{
+              "name": "Ex03",
+              "maxScore": 5
+            },{
+              "name": "Ex04",
+              "maxScore": 5
+            }
+          ]
+        }
+        ```
+        
+   2. Test case: `addgrade n/John Doe a/Ex04 s/5` <br>
+      Expected: Assignment `Ex04` with score `5` is added to `John Doe`. Details of the added grade shown in the status message.
+   
+   3. Test case: `addgrade/John Doe a/Ex05 s/5` <br>
+      Expected: No grade is added to John Doe. Details of the error message is shown in the status message.
+   
+   4. Other incorrect add grade commands to try: `addgrade`, `addgrade s/x` (where `x` is not a number)<br>
+      Expected: Similar to previous.
+   
+### Removing grade from a person
+1. Remove grade from a person
+    1. Prerequisites: Contact `John Doe` with grades(assignment name,score) `Ex01,5` and `Ex02,3` added.
+
+    2. Test case: `removegrade n/John Doe a/Ex01` <br>
+       Expected: Assignment `Ex01` is removed from `John Doe`. Details shown in the status message.
+
+    3. Test case: `removegrade/John Doe a/Ex05 s/5` <br>
+       Expected: No grade is removed from John Doe. Details of the error message is shown in the status message.
+
+    4. Other incorrect remove grade commands to try: `removegrade`, `removegrade a/x` (where `x` is any value)<br>
+       Expected: Similar to previous.
+   
+### Marking attendance of a person
+1. Marking attendance of a person in KonTActs
+   1. Test case: `mark n/John Doe w/5` <br>
+      Expected: `John Doe` is marked as present for week 5. Details of the marked week is shown in status message and the list.
+   
+   2. Test case: `mark n/John Doe w/s` <br>
+      Expected: No attendance is marked for `John Doe`. Details of the error is shown in the status message.
```

Comment:
The expected outcome seems confusing. We could probably say "Attendance would not be remarked for the same person and same week".